pkg/registry/blobs.go

```diff
@@ -44,10 +47,90 @@ func isBlob(req *http.Request) bool {
 		elem[len(elem)-2] == "uploads")
 }
 
+// blobHandler represents a blob storage backend.
+//
+// For all methods, repo is a string that can be passed to
+// pkg/name.NewRepository to validate the repository.
+type blobHandler interface {
+	// Stat returns the size of the blob, or ErrNotFound if the blob wasn't found.
+	Stat(repo string, h v1.Hash) (int64, error)
```

Comment:
If we want to extend this interface to manifests, it would be great if `Stat` returned a media type as well.

Similarly, if we take a `string` instead of a `v1.Hash` this would work for manifests.

Reply:
Ack. We'll cross that bridge when we get to it, uploads and manifests are next on the list.